web-app/src/pages/Profile/container.js

```diff
@@ -56,7 +60,20 @@ const ProfileContainer = Wrapped =>
             });
           });
         });
-      }
+      } 
+      if (
+        this.props.userDetails.employeeId !== null &&
+        this.state.contracts !== null
+      )
+        {   
+        getContractsByEmployeeId(this.props.userDetails.employeeId).then(response => {
```

Comment:
We already chatted about this, but I'm just making sure this isn't merged in yet by accident. Basically, we can't update state in `ComponentDidUpdate` or you'll be stuck in a loop.